File: python/problem001.py
def multiples():
    """returns sum of all multiples of 3 and 5 below 1000"""
    sum = 0
    x = 0
    while x < 1000:
        if x % 3 == 0:
            sum += x 
            x += 1
        elif x % 5 == 0:
            sum += x
            x += 1
        else:
            x += 1
    print(sum)

multiples() 

File: rust/problem001.rs
/// Calculate the sum of all multiples of 3 and 5 below n
fn multiples(n : i32) -> i32 {
    let mut sum = 0;
    let mut i = 3;
    while i < n {
        sum += i;
        i += 3;
    }

    i = 5;
    while i < n {
        sum += i;
        i += 5;
    }

    i = 15;
    while i < n {
        sum -= i;
        i += 15;
    }

    return sum;
}

fn main() {
    let sum : i32 = multiples(1000);
    println!("{}", sum);
}
